astropy/visualization/wcsaxes/patches.py

```diff
@@ -81,9 +83,15 @@ class SphericalCircle(Polygon):
 
     def __init__(self, center, radius, resolution=100, vertex_unit=u.degree, **kwargs):
 
-        # Extract longitude/latitude, either from a tuple of two quantities, or
-        # a single 2-element Quantity.
-        longitude, latitude = center
+        # Extract longitude/latitude, either from a SkyCoord object, or
+        # from a tuple of two quantities or a single 2-element Quantity.
+        if isinstance(center, SkyCoord):
```

Comment:
It is possible to define a `SkyCoord` using Cartesian coordinates. In that case an attempt to access longitude and latitude will not succeed because those are coordinates in a spherical system. It would therefore be better to include a check if `center.representation_type` is `SphericalRepresentation`.

Reply:
@eerovaher Thanks for the review! I have a doubt that should an error be raised if a SkyCoord using Cartesian coordinates is passed.

Reply:
If the `representation_type` of the `SkyCoord` is not `SphericalRepresentation` then maybe the best thing to do would be to convert the `SkyCoord` into spherical coordinates but also to raise a warning to make sure that the user knows what is happening.